flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -210,7 +210,7 @@
 
 	/** The exception that caused the job to fail. This is set to the first root exception
 	 * that was not recoverable and triggered job failure */
-	private volatile Throwable failureCause;
+	private final ErrorInfo failureCause = new ErrorInfo();
```

Comment:
making the fields of the `ErrorInfo` final and `failureCause` volatile is better imho.